components/brave_wallet_ui/page/screens/swap/hooks/useSwap.ts

```diff
@@ -131,15 +118,23 @@ export const useSwap = () => {
   const [slippageTolerance, setSlippageTolerance] = useState<string>('0.5')
   const [selectedGasFeeOption, setSelectedGasFeeOption] = useState<GasFeeOption>(gasFeeOptions[1])
   const [initialized, setInitialized] = useState<boolean>(false)
-  const [spotPrices, setSpotPrices] = useState<SpotPrices>({
-    fromAsset: '',
-    toAsset: '',
-    nativeAsset: ''
-  })
-  // FIXME(onyb): use global store
   const [tokenBalances, dispatchBalancesUpdate] = useReducer(balancesReducer, {})
   const [abortController, setAbortController] = useState<AbortController | undefined>(undefined)
 
+  const tokenPriceIds = useMemo(
+    () =>
+      [nativeAsset, fromToken, toToken]
+        .flatMap(token => (token ? [getPriceIdForToken(token)] : [])),
+    [nativeAsset, fromToken, toToken]
+  )
```

Comment:
How about this?

```suggestion
    const tokenPriceIds = useMemo(
    () =>
      [nativeAsset, fromToken, toToken]
        .filter(
          (token): token is BraveWallet.BlockchainToken => token !== undefined
        )
        .map((token) => getPriceIdForToken(token)),
    [nativeAsset, fromToken, toToken]
  )
```

Reply:
Fixed here: [271ccf9](https://github.com/brave/brave-core/pull/18919/commits/271ccf94559721a8e8a900c2a53f6725d858289c)